bc_obps/registration/models/operation_designated_operator_timeline.py

```diff
@@ -0,0 +1,35 @@
+from django.db import models
+from registration.models import (
+    Operation,
+    Operator,
+    TimeStampedModel,
+)
+from simple_history.models import HistoricalRecords
+
+
+class OperationDesignatedOperatorTimeline(TimeStampedModel):
+    operation = models.ForeignKey(Operation, on_delete=models.DO_NOTHING, related_name="designated_operators")
+    operator = models.ForeignKey(Operator, on_delete=models.DO_NOTHING, related_name="operation_designated_operators")
```

Comment:
We have used "DO_NOTHING" in many places, and I believe we should reconsider and change it to something more appropriate at some point. When we use "DO_NOTHING," we effectively bypass errors very gracefully. However, if we ever want to retrieve that record, we encounter a database error. I think we should prevent this before actually deleting it. For example, if we delete one of our operations for any reason, the record from this table associated with that operation remains unchanged. However, if we try to retrieve that record and want to fetch the operation of that record, the database will encounter an issue because that operation no longer exists. In short, we need to use "PROTECT" instead of "DO_NOTHING" in this table and in the other table related to the facility to prevent the deletion of the related records.
More about this: https://docs.djangoproject.com/en/5.1/ref/models/fields/#django.db.models.ForeignKey.on_delete